src/pages/TeacherDashboard.tsx

```
@@ -1,83 +1,133 @@
+          </p>
         </div>
       </div>
     );
   }
 
-  // filter only classes where this teacher is assigned
-  const assignedClasses = classes.filter(cls =>
-    cls.subjects.some(subject => subject.teacherId === user.id)
-  );
+  // Fetch all classes assigned to this teacher
+  useEffect(() => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)